src/app/components/splash-screen/splash-screen.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit, OnDestroy , ViewEncapsulation } from '@angular/core';
+import { UtilityService } from 'src/app/services/utility.service';
+import {IMAGE_TEXT} from "src/constants/constant";
+@Component({
+  selector: 'app-splash-screen',
+  templateUrl: './splash-screen.component.html',
+  styleUrls: ['./splash-screen.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SplashScreenComponent implements OnInit , OnDestroy {
+  windowWidth: string;
+  showSplash: boolean;
+  imgText: string;
+  timeOutIDs: number[] = [];
+  imageUrl = "../../../assets/icons";
+
+  constructor(private _utilityService: UtilityService) {}
+
+  ngOnInit() {
+    this.imgText = IMAGE_TEXT;
+    this.showSplash = this._utilityService.SHOW_SPLASH;
+    
+    this.timeOutIDs.push(
+      window.setTimeout (() => this.showSplash = false,2000)
+    );
```

Comment:
Why you are pushing the setTimeout into the array?

```suggestion
    this.timeOutIDs =   setTimeout(() => {
              this.showSplash = false
            }, 2000);
```

Reply:
Resolved.
```
Vaia, as far I've studied there might be more than one setTimeout in one .ts file. So good practice is holding all of them in an array and in the end destroy all using forEach. 
Though, it's not the case here as I've used only one setTimeout. I've just practiced  in my code. 
```